diff --git a/load-image.js b/load-image.js
index 6425f736adf5e57d22458630f75776fb5162822c..a77edd327c8c07333bb6bcf9c535be94095f1c59 100644
--- a/load-image.js
+++ b/load-image.js
@@ -62,14 +62,25 @@ function makeRequest(url, resolve, reject, redirectCount, requestOptions) {
 
   lib
     .get(url.toString(), requestOptions || {}, (res) => {
-      const shouldRedirect = REDIRECT_STATUSES.has(res.statusCode) && typeof res.headers.location === 'string'
-      if (shouldRedirect && redirectCount > 0)
-        return makeRequest(new URL(res.headers.location), resolve, reject, redirectCount - 1, requestOptions)
-      if (typeof res.statusCode === 'number' && (res.statusCode < 200 || res.statusCode >= 300)) {
-        return reject(new Error(`remote source rejected with status code ${res.statusCode}`))
-      }
+      try {
+        const shouldRedirect = REDIRECT_STATUSES.has(res.statusCode) && typeof res.headers.location === 'string'
+        if (shouldRedirect && redirectCount > 0) {
+          return makeRequest(
+            new URL(res.headers.location, url.origin),
+            resolve,
+            reject,
+            redirectCount - 1,
+            requestOptions,
+          )
+        }
+        if (typeof res.statusCode === 'number' && (res.statusCode < 200 || res.statusCode >= 300)) {
+          return reject(new Error(`remote source rejected with status code ${res.statusCode}`))
+        }
 
-      consumeStream(res).then(resolve, reject)
+        consumeStream(res).then(resolve, reject)
+      } catch (e) {
+        return reject(e)
+      }
     })
     .on('error', reject)
 }
